evalml/tests/automl_tests/test_automl.py

```diff
@@ -1251,7 +1251,7 @@ class DummyPipeline(dummy_binary_pipeline_class):
     np.testing.assert_equal(pipeline_results["percent_better_than_baseline_all_objectives"]['F1'], answer)
 
 
-@pytest.mark.parametrize("max_batches", [None, 1, 5, 8, 9, 10, 12])
+@pytest.mark.parametrize("max_batches", [None, 1, 5, 8, 9, 10, 12, 20])
 @patch('evalml.pipelines.BinaryClassificationPipeline.score', return_value={"Log Loss Binary": 0.8})
 @patch('evalml.pipelines.BinaryClassificationPipeline.fit')
 def test_max_batches_works(mock_pipeline_fit, mock_score, max_batches, X_y_binary):
```

Comment:
I think we should add a test that checks whether the ensembler is fit the correct number of times for different values of max_iterations. For example, I ran max_pipelines=49 and max_pipelines=50 and the ensembler only ran for max_pipelines=50. That's expected but might be good to document that in a test.

Reply:
Ah yes, I added max_batches=20 because I was testing with a subset. You're right, 49 and 50 are good numbers since right now, we have 8 pipeline classes, so the stacked ensemble runs for the first time after Baseline (1) + 8 pipelines in first batch (8) + 8 *5 for each pipeline class tuning round.

Reply:
(Gonna try making this dynamic since this number will update when we add more estimators 🤔 ) Thanks for the great suggestion!